core/src/main/java/com/linecorp/armeria/client/circuitbreaker/DefaultCircuitBreakerClientHandler.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.Request;
+
+final class DefaultCircuitBreakerClientHandler<I extends Request> implements CircuitBreakerClientHandler<I> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultCircuitBreakerClientHandler.class);
+
+    static <I extends Request> CircuitBreakerClientHandler<I> of(CircuitBreaker cb) {
+        return of((ctx, req) -> cb);
+    }
+
+    static <I extends Request> DefaultCircuitBreakerClientHandler<I> of(CircuitBreakerMapping cb) {
+        return new DefaultCircuitBreakerClientHandler<>(cb);
+    }
```

Comment:
How about moving these factories to `CircuitBreakerClientHandler`?

Reply:
Good point! Done!